bson/src/main/com/mongodb/lang/Nullable.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.lang;
+
+import javax.annotation.Nonnull;
```

Comment:
Just to clarify using a Java 8 annotation won't have any impact / errors for Java 6/7 users? Because its removed from the jar at compile time and is just kept in the source jar for tools such as intellij to use.

If so ship it!

Reply:
The annotations are retained at runtime:

    @Retention(RetentionPolicy.RUNTIME)

so that tools like the Kotlin compiler and the Spotbugs analyzer can use them without requiring the original source.  The Kotlin compiler, in particular, can recognize the javax.annotation.Nonnull meta-annotation defined in JSR-305 (but never actually included in any JDK) without requiring a JSR-305 implementation on the class path.  See https://kotlinlang.org/docs/reference/java-interop.html#jsr-305-support for a description.

I didn't use the @UnderMigration meta-annotation described there, but we could add that as well.